### PR TITLE
Fixed: (HttpIndexerBase) Improve error messages on request timeouts and network issues

### PR DESCRIPTION
**Pull request by mynameisbogdan**
#### Database Migration
NO

/cc @bakerboy448

### CODE REVIEW

File: src/NzbDrone.Core/Indexers/HttpIndexerBase.cs [hunk 1]
```diff
@@ -258,6 +259,16 @@ protected virtual async Task<IndexerPageableQueryResult> FetchReleases(Func<IInd
                 _indexerStatusService.RecordFailure(Definition.Id);
                 _logger.Warn(ex, "{0}", url);
             }
+            catch (HttpRequestException ex)
+            {
+                _indexerStatusService.RecordFailure(Definition.Id);
+                _logger.Warn(ex, "Unable to connect to indexer, please check your DNS settings. {0}", url);
```
Comment (bakerboy448) [inline src/NzbDrone.Core/Indexers/HttpIndexerBase.cs:265]:
```suggestion
                _logger.Warn(ex, "Unable to connect to indexer, please check your DNS settings and ensure IPv6 is working or disabled. [{0}]", url);
```

File: src/NzbDrone.Core/Indexers/HttpIndexerBase.cs [hunk 2]
```diff
@@ -485,12 +496,22 @@ protected virtual async Task<ValidationFailure> TestConnection()
...
-                }
+                _logger.Warn(ex, "Unable to connect to indexer");
+
+                return new ValidationFailure(string.Empty, "Unable to connect to indexer, check the log for more details");
+            }
+            catch (HttpRequestException ex)
+            {
+                _logger.Warn(ex, "Unable to connect to indexer");
+
+                return new ValidationFailure(string.Empty, "Unable to connect to indexer, please check your DNS settings. " + ex.Message);
```
Comment (bakerboy448) [inline src/NzbDrone.Core/Indexers/HttpIndexerBase.cs:508]:
```suggestion
                return new ValidationFailure(string.Empty, "Unable to connect to indexer, please check your DNS settings and ensure IPv6 is working or disabled. " + ex.Message);
```

File: src/NzbDrone.Core/Indexers/HttpIndexerBase.cs [hunk 3]
```diff
@@ -485,12 +496,22 @@ protected virtual async Task<ValidationFailure> TestConnection()
...
+            }
+            catch (TaskCanceledException ex)
+            {
+                _logger.Warn(ex, "Unable to connect to indexer");
```
Comment (bakerboy448) [inline src/NzbDrone.Core/Indexers/HttpIndexerBase.cs:512]:
```suggestion
                _logger.Warn(ex, "Unable to connect to indexer due to timeout.");
```

File: src/NzbDrone.Core/Indexers/HttpIndexerBase.cs [hunk 4]
```diff
@@ -258,6 +259,16 @@ protected virtual async Task<IndexerPageableQueryResult> FetchReleases(Func<IInd
...
             }
+            catch (HttpRequestException ex)
+            {
+                _indexerStatusService.RecordFailure(Definition.Id);
+                _logger.Warn(ex, "Unable to connect to indexer, please check your DNS settings. {0}", url);
+            }
+            catch (TaskCanceledException ex)
+            {
+                _indexerStatusService.RecordFailure(Definition.Id);
+                _logger.Warn(ex, "Unable to connect to indexer, possibly due to a timeout. {0}", url);
```
Comment (bakerboy448) [inline src/NzbDrone.Core/Indexers/HttpIndexerBase.cs:270]:
```suggestion
                _logger.Warn(ex, "Unable to connect to indexer due to timeout. [{0}]", url);
```

File: src/NzbDrone.Core/Indexers/HttpIndexerBase.cs (hunk 1 above)
Comment (bakerboy448) [inline src/NzbDrone.Core/Indexers/HttpIndexerBase.cs:265]:
```suggestion
                _logger.Warn(ex, "Unable to connect to indexer, please check your networking. {0}", url);
```

File: src/NzbDrone.Core/Indexers/HttpIndexerBase.cs [hunk 5]
```diff
@@ -485,12 +496,22 @@ protected virtual async Task<ValidationFailure> TestConnection()
...
+            {
+                _logger.Warn(ex, "Unable to connect to indexer");
+
+                return new ValidationFailure(string.Empty, "Unable to connect to indexer, please check your DNS settings. " + ex.Message);
+            }
+            catch (TaskCanceledException ex)
+            {
+                _logger.Warn(ex, "Unable to connect to indexer");
+
+                return new ValidationFailure(string.Empty, "Unable to connect to indexer, possibly due to a timeout. If needed try again. " + ex.Message);
```
Comment (bakerboy448) [inline src/NzbDrone.Core/Indexers/HttpIndexerBase.cs:514]:
```suggestion
                return new ValidationFailure(string.Empty, "Unable to connect to indexer due to timeout and no response after 100 seconds. Try again or check your network settings. " + ex.Message);
```

**Review by bakerboy448 — CHANGES_REQUESTED** [review]
Defer [X] to Qstick

for the logging could this be caught and logged for all HTTP requests App Wide for both issues and then just leave the validation errors in indexer http?

File: src/NzbDrone.Core/Indexers/HttpIndexerBase.cs (hunk 5 above)
Comment (Qstick) [reply src/NzbDrone.Core/Indexers/HttpIndexerBase.cs]: Agree with bogdan, let's leave it generic

**Review by Qstick — APPROVED** [review]

**Pull request merged**